packages/shared/profiles/sagas.ts

```diff
@@ -183,52 +197,91 @@ export function* handleFetchProfile(action: ProfileRequestAction): any {
-        return null
-      }
+  if (!identity) throw new Error("Can't fetch profile if there is no ExplorerIdentity")
 
-      avatar.hasClaimedName = !!avatar.name && avatar.hasClaimedName // old lambdas profiles don't have claimed names if they don't have the "name" property
-      avatar.hasConnectedWeb3 = true
+  try {
+    const userIdsToFetch = userIds.filter((userId) => userId.toLowerCase() !== identity.address.toLowerCase())
+    const isFetchingOwnUser = userIds.some((userId) => userId.toLowerCase() === identity.address.toLowerCase())
+
+    const avatars: Avatar[] = yield (!isFetchingOwnUser && call(getRemoteProfiles, userIdsToFetch, version)) || []
 
-      return avatar
+    const ownProfile = isFetchingOwnUser && (yield call(readProfileFromLocalStorage))
+    if (ownProfile != null) {
+      avatars.push(ownProfile)
     }
+
+    yield put(profilesSuccess(avatars))
   } catch (error: any) {
-    if (error.message !== 'Profile not found') {
-      defaultLogger.log(`Error requesting profile for auth check ${userId}, `, error)
+    trackEvent('error', {
+      context: 'kernel#saga',
+      message: `Error requesting profiles for ${userIds}: ${error}`,
+      stack: error.stack || ''
+    })
+    yield put(profilesFailure(userIds, `${error}`))
+  }
+}
+
+function* getRemoteProfiles(userIds: string[], version?: number) {
+  try {
+    const profiles: RemoteProfile[] = (yield call(profilesServerRequest, userIds, version)) as any
+
+    let avatars = profiles.map((profile) => {
+      let avatar = profile.avatars[0]
+
+      if (avatar) {
+        avatar = ensureAvatarCompatibilityFormat(avatar)
+        if (!validateAvatar(avatar)) {
+          defaultLogger.warn(`Remote avatar for users is invalid.`, userIds, avatar, validateAvatar.errors)
+          return null
+        }
+
+        avatar.hasClaimedName = !!avatar.name && avatar.hasClaimedName // old lambdas profiles don't have claimed names if they don't have the "name" property
+        avatar.hasConnectedWeb3 = true
+
+        return avatar
+      }
+    })
+
+    return avatars
+  } catch (error: any) {
+    if (error.message !== 'Profiles not found') {
+      defaultLogger.log(`Error requesting profiles for auth check ${userIds}, `, error)
     }
   }
   return null
 }
 
-export async function profileServerRequest(userId: string, version?: number) {
+export async function profilesServerRequest(userIds: string[], version?: number): Promise<RemoteProfile[]> {
   const state = store.getState()
   const catalystUrl = getCatalystServer(state)
 
   try {
-    let url = `${catalystUrl}/lambdas/profiles?id=${userId}`
+    let url = `${catalystUrl}/lambdas/profiles`
     if (version) url = url + `&version=${version}`
 
-    const response = await fetch(url)
+    const response = await fetch(url, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({ ids: userIds })
```

Comment:
Version is _very_ important to kernel and renderer. Please account for that, it is a caching mechanism